main.go

```diff
@@ -216,6 +240,26 @@ func reconcileBalancer(ctx context.Context, bal balancer.Balancer, c cluster.Clu
 	}
 }
 
+// Validate new events and pass them to state machine
+func reconcileMachine(ctx context.Context, c cluster.Cluster, signer events.Signer, machine *state.Machine) error {
```

Comment:
1. Where do you distribute the state (or replay the events) when a new Catalyst node joins? How do you make it aware of the current state in the cluster?

2. Do we have any persistence for the state machine? What if all Catalyst nodes are down?